saleor/graphql/shipping/types.py

```diff
@@ -76,3 +80,7 @@ def resolve_countries(root: models.ShippingZone, *_args):
     @staticmethod
     def resolve_shipping_methods(root: models.ShippingZone, *_args):
         return root.shipping_methods.all()
+
+    @staticmethod
+    def resolve_warehouses(root: models.ShippingZone, *_args):
```

Comment:
In top-level query this field requires `MANAGE_PRODUCTS` so it should be required here as well.